server/supernova/server/server.cpp

```diff
@@ -269,27 +269,43 @@ static bool set_realtime_priority(int thread_index)
 #ifdef NOVA_TT_PRIORITY_RT
 
 #ifdef JACK_BACKEND
+#ifndef _WIN32
         int priority = instance->realtime_priority();
         if (priority >= 0)
             success = true;
-
+#else
+		int priority = thread_priority_interval_rt().second;
+		success = true;
+#endif
 #elif _WIN32
         int priority = thread_priority_interval_rt().second;
+		success = true;
 #else
         int min, max;
         boost::tie(min, max) = thread_priority_interval_rt();
         int priority = max - 3;
         priority = std::max(min, priority);
 #endif
 
-        if (success)
+        if (success){
+			//std::cout << "setting thread priority " << priority << std::endl;
             success = thread_set_priority_rt(priority);
+		}
 #endif
     }
 
-    if (!success)
-        std::cout << "Warning: cannot raise thread priority" << std::endl;
+    if (!success){
+#ifdef _WIN32
+		std::cout << "win32 error setting thread priority " << std::endl;
+		char *s;
+        FormatMessage( FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                       nullptr, GetLastError() , MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (char*)&s, 0, NULL );
 
+        std::cout << "*** ERROR: GetProcAddress err " << s << std::endl;
+        LocalFree( s );
+#endif
+        std::cout << "Warning: cannot raise thread priority" << std::endl;
+	}
```

Comment:
I'm finding the added logic here very difficult to follow; you should spend some time refactoring it so that it's clearer. ideally have a separate function for each case. also, please add comments to #else/#endif directives so the condition is clear. (i.e. the endif on line 306 should read `#endif // _WIN32`.

finally, please add a space between closing paren and open bracket (line 297, line 290)

Reply:
comments added